From b3881a330888acc429b83e6432583c809b189dd2 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:41:58 +0800
Subject: [PATCH 41/60] GSI Android 17 compatibility:
 0043-Some-Samsung-devices-use-SW_FLIP-0x15-rather-than-SW.patch

Change-Id: Ia03746c49518d7476bb1693658b61fd34751ebd6
---
 .../server/input/InputManagerService.java     | 25 +++++++++++++------
 1 file changed, 18 insertions(+), 7 deletions(-)

diff --git a/services/core/java/com/android/server/input/InputManagerService.java b/services/core/java/com/android/server/input/InputManagerService.java
index 40d23237e..f58f4379c 100644
--- a/services/core/java/com/android/server/input/InputManagerService.java
+++ b/services/core/java/com/android/server/input/InputManagerService.java
@@ -460,6 +460,8 @@ public class InputManagerService extends IInputManager.Stub
 
     /** Switch code: Cover. When set, the cover is closed. */
     public static final int SW_MACHINE_COVER = 0x10;
+    // Samsung
+    public static final int SW_FLIP = 0x15;
 
     public static final int SW_LID_BIT = 1 << SW_LID;
     public static final int SW_TABLET_MODE_BIT = 1 << SW_TABLET_MODE;
@@ -628,13 +630,22 @@ public class InputManagerService extends IInputManager.Stub
     }
 
     private void checkForSwMachineCover() {
-        int machineCoverState = getSwitchState(-1 /* deviceId */, InputDevice.SOURCE_ANY, SW_MACHINE_COVER);
-        if (machineCoverState != KEY_STATE_UNKNOWN) {
-            android.util.Log.e("PHH", "Found a SW_MACHINE_COVER. Use this instead of SW_LID");
-            mSwForLid = SW_MACHINE_COVER;
-        } else {
-            android.util.Log.e("PHH", "Not found a SW_MACHINE_COVER");
-        }
+            int machineCoverState = getSwitchState(-1, InputDevice.SOURCE_ANY, SW_MACHINE_COVER);
+            if (machineCoverState != KEY_STATE_UNKNOWN) {
+                android.util.Log.e("PHH", "Found a SW_MACHINE_COVER. Use this instead of SW_LID");
+                mSwForLid = SW_MACHINE_COVER;
+            } else {
+                android.util.Log.e("PHH", "Not found a SW_MACHINE_COVER");
+            }
+
+            int flip = getSwitchState(-1, InputDevice.SOURCE_ANY, SW_FLIP);
+            if (flip != KEY_STATE_UNKNOWN) {
+                android.util.Log.e("PHH", "Found a SW_FLIP. Use this instead of SW_LID");
+                mSwForLid = SW_FLIP;
+            } else {
+                android.util.Log.e("PHH", "Not found a SW_FLIP");
+            }
+            android.util.Log.e("PHH", "Switch state got " + machineCoverState + " and " + flip);
     }
 
     void registerLidSwitchCallbackInternal(@NonNull LidSwitchCallback callback) {
